src/main/kotlin/com/sourcegraph/cody/config/SettingsMigration.kt

```diff
@@ -83,6 +85,40 @@ class SettingsMigration : Activity {
     migrateEnterpriseAccount(project, requestExecutorFactory, customRequestHeaders)
   }
 
+  private fun migrateLlms(project: Project) {
+    HistoryService.getInstance(project)
+        .state
+        .chats
+        .filter { it.llm == null }
+        .forEach {
+          val (model, provider, title) =
+              modelToProviderAndTitle.getOrDefault(it.model, Triple("", "Unknown", "Unknown"))
```

Comment:
That is my only doubt. I think you should stick to the `null`s.
Then this code will work properly:

```kt
val modelFromState =
             state.llm?.let {
               ChatModelsResponse.ChatModelProvider(
                   default = it.model == null,
                   codyProOnly = false,
                   provider = it.provider ?: "Unknown",
                   title = it.title ?: "Unknown",
                   model = it.model ?: "")
             }
```

Reply:
will be fixed in https://github.com/sourcegraph/jetbrains/pull/753